src/client/settings/org-webhook-list.ts

```diff
@@ -28,22 +28,35 @@ async function getState() {
 }
 
 function hookTemplate(org: OrgWebHookState) {
-  // TODO: No idea how to add disabled state (i.e. bring
-  // org.viewerCanAdminister to checkbox)
+  const checkboxClick = () => {
+    console.log('Checkbox Click: ', org);
+  };
   return html`
   <div class="settings-toggle-item">
-    <input class="settings-toggle-item__toggle" type="checkbox"></input>
+    <input class="settings-toggle-item__toggle" type="checkbox" on-click="${() => checkboxClick()}"></input>
     <div class="settings-toggle-item__details">
       <h6>${org.name}</h6>
       <p>TODO: Add state message</p>
     </div>
   </div>`;
 }
 
+function requestPermissionTemplate() {
+  const readOrgsClick = () => {
+    window.location.href = `/oauth.html?scope=read:org&final-redirect=${window.location.href}`;
+  };
+  const requestPermissionTemplate = html`<p><button on-click="${() => readOrgsClick()}">Enable Org Reading</button></p>`;
+  return requestPermissionTemplate;
+}
+
 async function updateUI() {
-  const state = await getState();
-  const orgTemplate = html`${state.orgs.map(hookTemplate)}`;
-  render(orgTemplate, orgListContainer);
+  try {
+    const state = await getState();
```

Comment:
Why is this a try/catch? Does  `getState()` throw when it doesn't have permission to read?

Reply:
Yes - if the API can't request the orgs from github, our server returns a 500, which throws the fetch  in `getState` . I can catch in getState() and return null or return an object with `readOrgGranted: false` and true otherwise.

This decision was made along the lines of - this is a bad / undesirable situation to be in.